### PR TITLE
update palantirJavaFormat for Java 21

### PR DESCRIPTION
**Pull request by slachiewicz**
Based on @basil user report and fix https://github.com/diffplug/spotless/issues/1774

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 ## [Unreleased]
 ### Added
 * Support Ktlint 1.0.0 ([#1808](https://github.com/diffplug/spotless/pull/1808)).
+* Added support to run with Java 21 ([#1774](https://github.com/diffplug/spotless/issiue/1774)).
```
Comment (HTHou) [inline CHANGES.md:15]:
```suggestion
* Added support to run with Java 21 ([#1774](https://github.com/diffplug/spotless/issue/1774)).
```

File: CHANGES.md [hunk 2]
```diff
@@ -16,6 +16,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
   the version accordingly. ([#1804](https://github.com/diffplug/spotless/issues/1804)).
 ### Fixed
 * Added support for plugins when using Prettier version `3.0.0` and newer. ([#1802](https://github.com/diffplug/spotless/pull/1802))
+* Added support to run with Java 21 ([#1774](https://github.com/diffplug/spotless/issue/1774)).
```
Comment (slawekjaranowski) [inline CHANGES.md:19]: https://github.com/diffplug/spotless/issues/1774

issue -> issues 😄

Comment (slawekjaranowski) [reply CHANGES.md:20]: still is `issue` - should be `issues` in link 😄

Comment (slachiewicz) [reply CHANGES.md:20]: I should learn more about merges ;-) thx Slawek

**Comment by nedtwigg** [comment]
Looks like issues around gradle variants.

```
 project : > com.palantir.javaformat:palantir-java-format:2.38.0 > com.palantir.javaformat:palantir-java-format-spi:2.38.0

  Caused by: org.gradle.internal.component.AmbiguousConfigurationSelectionException: The consumer was configured to find attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external'. However we cannot choose between the following variants of com.google.guava:guava:32.1.2-jre:
    - androidRuntimeElements
    - jreRuntimeElements
  All of them match the consumer attributes:
    - Variant 'androidRuntimeElements' capabilities com.google.collections:google-collections:32.1.2-jre and com.google.guava:guava:32.1.2-jre declares attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external':
        - Unmatched attributes:
            - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
            - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
            - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
            - Provides org.gradle.status 'release' but the consumer didn't ask for it
            - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
    - Variant 'jreRuntimeElements' capabilities com.google.collections:google-collections:32.1.2-jre and com.google.guava:guava:32.1.2-jre declares attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external':
        - Unmatched attributes:
            - Provides org.gradle.jvm.environment 'standard-jvm' but the consumer didn't ask for it
            - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
            - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
            - Provides org.gradle.status 'release' but the consumer didn't ask for it
            - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
      at app//com.diffplug.spotless.java.PalantirJavaFormatStepTest.jvm13Features(PalantirJavaFormatStepTest.java:35)
```

We had similar issues in https://github.com/diffplug/spotless/pull/1808

**Comment by nedtwigg** [comment]
FYI, `main` is now on Gradle 8.4, so if you want to bring Java 21 back into the test matrix, that would be welcome. I don't think it will help with the variant issue above though...

**Comment by slachiewicz** [comment]
Unfortunately, it's still the same error with the new Gradle - no idea how to fix this dependency issue.

**Comment by nedtwigg** [comment]
Anyone know why this PR isn't running CI? It's doing `spotlessCheck assemble testClasses` but none of the actual testrunner runs...

**Comment by nedtwigg** [comment]
```
  org.gradle.api.artifacts.ResolveException: Could not resolve all dependencies for [com.palantir.javaformat:palantir-java-format:2.38.0].
      at com.diffplug.spotless.java.PalantirJavaFormatStepTest.jvm13Features(PalantirJavaFormatStepTest.java:35)
  Caused by: org.gradle.api.internal.artifacts.ivyservice.DefaultLenientConfiguration$ArtifactResolveException: Could not resolve all files for configuration ':detachedConfiguration1'.
      at app//com.diffplug.spotless.java.PalantirJavaFormatStepTest.jvm13Features(PalantirJavaFormatStepTest.java:35)
  Caused by: org.gradle.internal.resolve.ModuleVersionResolveException: Could not resolve com.google.guava:guava:32.1.2-jre.
  Required by:
      project : > com.palantir.javaformat:palantir-java-format:2.38.0
      project : > com.palantir.javaformat:palantir-java-format:2.38.0 > com.palantir.javaformat:palantir-java-format-spi:2.38.0

  Caused by: org.gradle.internal.component.AmbiguousConfigurationSelectionException: The consumer was configured to find attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external'. However we cannot choose between the following variants of com.google.guava:guava:32.1.2-jre:
    - androidRuntimeElements
    - jreRuntimeElements
  All of them match the consumer attributes:
    - Variant 'androidRuntimeElements' capabilities com.google.collections:google-collections:32.1.2-jre and com.google.guava:guava:32.1.2-jre declares attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external':
        - Unmatched attributes:
            - Provides org.gradle.jvm.environment 'android' but the consumer didn't ask for it
            - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
            - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
            - Provides org.gradle.status 'release' but the consumer didn't ask for it
            - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
    - Variant 'jreRuntimeElements' capabilities com.google.collections:google-collections:32.1.2-jre and com.google.guava:guava:32.1.2-jre declares attribute 'org.gradle.category' with value 'library', attribute 'org.gradle.dependency.bundling' with value 'external':
        - Unmatched attributes:
            - Provides org.gradle.jvm.environment 'standard-jvm' but the consumer didn't ask for it
            - Provides org.gradle.jvm.version '8' but the consumer didn't ask for it
            - Provides org.gradle.libraryelements 'jar' but the consumer didn't ask for it
            - Provides org.gradle.status 'release' but the consumer didn't ask for it
            - Provides org.gradle.usage 'java-runtime' but the consumer didn't ask for it
      at app//com.diffplug.spotless.java.PalantirJavaFormatStepTest.jvm13Features(PalantirJavaFormatStepTest.java:35)
```

**Comment by Goooler** [comment]
@slachiewicz Seems I've fixed the dependency resolving issue, you can ref https://github.com/Goooler/spotless/pull/12. And if you wanna run CI on Java 21, still need to fix GJF tests.

**Comment by StefanLobbenmeierObjego** [comment]
FYI with gradle 8.5 being released today this became more urgent as it blocks users from running gradle on java 21

**Comment by Goooler** [comment]
It's in #1920.

**Comment by nedtwigg** [comment]
- Going to be merged soon via #1920

**Pull request closed without merging**